src/js/controllers/chargeMoneyController.ts

```diff
@@ -0,0 +1,31 @@
+import ChargeMoneyView from '../views/chargeMoneyView';
+import VendingMachine from '../vendingMachine/vendingMachine';
+import { CUSTOM_EVENT } from '../constants/appContants';
+import { ChargeMoneyDetailType } from '../types/types';
+
+export default class ChargeMoneyController {
+  vendingMachine: VendingMachine;
+  chargeMoneyView: ChargeMoneyView;
+
+  constructor(vendingMachine: VendingMachine) {
+    this.vendingMachine = vendingMachine;
+    this.chargeMoneyView = new ChargeMoneyView();
+
+    window.addEventListener(CUSTOM_EVENT.CHARGE_MONEY, this.handleChargeMoney.bind(this));
+  }
+
+  loadPage() {
+    const { coins, money } = this.vendingMachine;
+
+    this.chargeMoneyView.render(coins, money);
+  }
+
+  handleChargeMoney(event: CustomEvent) {
+    const { inputMoney }: ChargeMoneyDetailType = event.detail;
```

Comment:
```suggestion
  handleChargeMoney(event: CustomEvent<ChargeMoneyDetailType>) {
    const { inputMoney } = event.detail;
```

CustomEvent에 제네릭으로 넣어주면 event.detail이 넣어준 타입으로 받아줄 수 있어요.
제네릭을 지정안했을때의 기본값이 CustomEvent<any>이기 때문에 동작은 하지만 any를 최소화하는걸 좋아해서 추천드려요.

혹시 변경하신다면 다른 CustomEvent에도 적용부탁드릴게요 🙂

Reply:
타입 스크립트 공부 많이 해야겠네요 😭  컨트롤러 없애면서 다른 곳에도 적용할게 있는지 찾아보겠습니다!